### PR TITLE
feat: integrate form template with Gutenberg blocks 

### LINKED ISSUE #4582: feat: integrate form template with Gutenberg blocks
# Feature Request

## Developer Story

As a developer, I want to integrate form template with Gutenberg block so that admin can use it.

## Acceptance Criteria
ref:
https://www.loom.com/share/c82b8da3d62c433f9f742b004802b0b0
https://github.com/impress-org/givewp/pull/4567

- [x] Overlay options on Give Form block have been removed.
- [x] If form template other then legacy active for a form then admin can see only form change option in block setting. If legacy template active then admin can see old settings.
- [x] Update form selector screen in Give Form block
   - [x] Change selector size [ it is too big now]
   - [x] Change the background color to grayish.
   - [x] Remove `Add new form` button and form must select automatically after choosing a form.
image

### PR DESCRIPTION
**Pull request by ravinderk**
## Description
This Pr will resolve #4582

## How Has This Been Tested?
I tested this or by creating Give Form blocks and I also tested this with `Gutenberg 7.8.0`.

HukV1cmTAM

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: blocks/utils/index.js [hunk 1]
```diff
@@ -5,3 +5,56 @@
...
+ * Convert forms object in option
+ *
+ * @since 2.7.0
+ *
+ * @param {object} forms
+ * @param {number} SelectedFormId
+ *
+ * @return {boolean}
+ */
+export function isShowOldSettings( forms, SelectedFormId ) {
```
Comment (JasonTheAdams) [inline blocks/utils/index.js]: Minor detail, but this function name is awkward and the description doesn't help. Is it returning whether a form is legacy or not?

File: blocks/utils/index.js [hunk 2]
```diff
@@ -5,3 +5,56 @@
...
+	if ( forms ) {
+		const data = forms.find(
+			( form ) => {
+				return parseInt( form.id ) === parseInt( SelectedFormId );
+			}
+		);
```
Comment (JasonTheAdams) [inline blocks/utils/index.js]:
```suggestion
		const data = forms.find( form => parseInt( form.id ) === parseInt( SelectedFormId ) );
```

File: blocks/utils/index.js (hunk 1 above)
Comment (ravinderk) [reply blocks/utils/index.js]: fixed: https://github.com/impress-org/givewp/compare/a7a47c648d73...2d98c0e71114

**Comment by DevinWalker** [comment]
Pretty darn close to what I'm looking for @ravinderk - here's my review: https://www.loom.com/share/c569c68202bf42b0b09fd4a006bb58ff

I'll make the minor adjustments and merge.

**Pull request merged**